Sources/X509/X509BaseTypes/TimeCalculations.swift

```diff
@@ -0,0 +1,292 @@
+//
+// This source file is part of the SwiftCertificates open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftCertificates project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// This file contains code derived from the musl implementation at
+// https://git.musl-libc.org/cgit/musl/tree/src/time/__secs_to_tm.c and
+// https://git.musl-libc.org/cgit/musl/tree/src/time/__tm_to_secs.c.
+//
+// These implementations have been translated into Swift appropriately for this
+// use-case.
+//
+// The copyright for the original implementation is:
+//
+//----------------------------------------------------------------------
+// Copyright © 2005-2020 Rich Felker, et al.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//----------------------------------------------------------------------
+
+extension Int64 {
+    // 2000-03-01 (mod 400 year, immediately after feb29
+    @inlinable
+    static var leapoch: Int64 {
+        .secondsFromEpochToYear2000 + .secondsPerDay * (31 + 29)
+    }
+
+    @inlinable
+    static var secondsFromEpochToYear2000: Int64 {
+        946684800
+    }
+
+    @inlinable
+    static var daysPer400Years: Int64 {
+        365 * 400 + 97
+    }
+
+    @inlinable
+    static var daysPer100Years: Int64 {
+        365 * 100 + 24
+    }
+
+    @inlinable
+    static var daysPer4Years: Int64 {
+        365 * 4 + 1
+    }
+
+    @inlinable
+    static var secondsPerDay: Int64 {
+        24 * 60 * 60
+    }
+
+    @inlinable
+    static var secondsPerYear: Int64 {
+        .secondsPerDay * .daysPerYear
+    }
+
+    @inlinable
+    static var daysPerYear: Int64 {
+        365
```

Comment:
If we count February as 29 days long, shouldn't it be set to 366 days?

Reply:
Nope: we account for the leap years in this calculation elsewhere.